New sensor platform integration for Orange and Rockland Utility smart energy meter

bvlaicu opened the pull request:
## Description:


New sensor platform integration for [Orange and Rockland Utility](https://oru.com) smart energy meter.
Orange and Rockland Utility is an energy provider in NY and NJ, USA.
The sensor provides the last energy meter read in kWh.

## Docs PR:
https://github.com/home-assistant/home-assistant.io/pull/10725

## Example entry for `configuration.yaml`:
```yaml
sensor:
  - platform: oru
    meter_number: 701138804
```

The meter number can be found on the energy bill, on the top left corner, alongside the account number. 

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

bvlaicu committed:
New sensor platform integration for Orange and Rockland Utility smart energy meter

bvlaicu committed:
New sensor platform integration for Orange and Rockland Utility smart energy meter

homeassistant commented:
Hi @bvlaicu, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#27571).

Once you do that we will be able to review and accept this pull request.

Thanks!

bvlaicu committed:
Merge branch 'oru' of https://github.com/bvlaicu/home-assistant into oru

bvlaicu committed:
bumped the oru py version to 0.1.9

balloob commented on homeassistant/components/oru/sensor.py:
```diff
@@ -0,0 +1,69 @@
+SENSOR_ICON = "mdi:counter"
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
```
Please add a PLATFORM_SCHEMA to validate the meter number

bvlaicu replied:
Done.

balloob commented on homeassistant/components/oru/sensor.py:
```diff
@@ -0,0 +1,69 @@
+            self._state = last_read
+            self._available = True
+
+            _LOGGER.info("%s = %s %s", self.name, self._state, self.unit_of_measurement)
```
Le'ts not log this as info.

bvlaicu replied:
Done.

bvlaicu committed:
Added PLATFORM_SCHEMA
Adde unique_id property
Changed logger level from info to debug when printing the updated sensor value
Set the SCAN_INTERVAL to 15 mins
Added exception handling durin init when creating the oru meter instance

MartinHjelmare commented on homeassistant/components/oru/manifest.json:
```diff
@@ -0,0 +1,8 @@
+{
+    "domain": "oru",
+    "name": "Orange and Rockland Utility Smart Energy Meter Sensor",
+    "documentation": "https://www.home-assistant.io/components/",
```
Please link to the correct docs page of the integration.

bvlaicu replied:
Done

MartinHjelmare commented on homeassistant/components/oru/sensor.py:
```diff
@@ -0,0 +1,92 @@
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the sensor platform."""
+
+    meter_number = str(config.get(CONF_METER_NUMBER, None))
```
Please use `dict[key]` for required config keys and keys with default config schema values.

MartinHjelmare replied:
No need to copy to str since the option is validated already by the config schema.

bvlaicu replied:
Done

MartinHjelmare commented on homeassistant/components/oru/sensor.py:
```diff
@@ -0,0 +1,92 @@
+    try:
+        meter = Meter(meter_number)
+
+    except MemoryError:
```
Is this the correct exception?

bvlaicu replied:
Typo. Thanks! Fixed.

MartinHjelmare commented on homeassistant/components/oru/sensor.py:
```diff
@@ -0,0 +1,92 @@
+
+    add_entities([CurrentEnergyUsageSensor(meter)])
+
+    _LOGGER.debug("meter_number = %s", meter_number)
```
We start logging messages with capital letter.

Maybe we don't need to log the config option? We should know what we entered in the config.

bvlaicu replied:
Done. Started with capital letter. I'd like to leave this in, just in case. It's` debug` anyway.

MartinHjelmare commented on homeassistant/components/oru/sensor.py:
```diff
@@ -0,0 +1,92 @@
+            self._available = True
+
+            _LOGGER.debug(
+                "%s = %s %s", self.name, self._state, self.unit_of_measurement
```
Please start logging messages with capital letter (we don't end with period).

bvlaicu replied:
It does start with a capital. `self.name` starts with capital. Sample log message: 
```ORU Current Energy Usage = 0.132 kWh```

MartinHjelmare commented on homeassistant/components/oru/sensor.py:
```diff
@@ -0,0 +1,92 @@
+        except MeterError as err:
+            self._available = False
+
+            _LOGGER.error("Unexpected oru meter error: %s", str(err))
```
We don't need to copy to str. The exception argument has a `__str__` magic method.

bvlaicu replied:
Done.

MartinHjelmare reviewed (changes requested):
Looks good! Some minor comments.

bvlaicu committed:
Various fixes base on the PR review
+ Added SCAN_INTERVAL for 15 mins

MartinHjelmare commented on homeassistant/components/oru/manifest.json:
```diff
@@ -0,0 +1,8 @@
+{
+    "domain": "oru",
+    "name": "Orange and Rockland Utility Smart Energy Meter Sensor",
+    "documentation": "https://www.home-assistant.io/components/oru",
```
We've recently renamed the folder structure from `components` to `integrations`.

bvlaicu replied:
Done.

bvlaicu committed:
fixed path to documentation

MartinHjelmare reviewed (approved):
Good!

bvlaicu commented:
> Good!

@MartinHjelmare Martin, thanks for your review. Is there anything else I need to do for this new integration or is this pretty much it? Thanks.
